src/workerd/api/worker-rpc.c++

```diff
@@ -147,12 +186,107 @@ private:
   kj::ForkedPromise<void> promise;
 };
 
+// Given a value, check if it has a dispose method and, if so, invoke it.
+void tryCallDisposeMethod(jsg::Lock& js, jsg::JsValue value) {
+  js.withinHandleScope([&]() {
+    KJ_IF_SOME(obj, value.tryCast<jsg::JsObject>()) {
+      auto dispose = obj.get(js, js.symbolDispose());
+      if (dispose.isFunction()) {
+        jsg::check(v8::Local<v8::Value>(dispose).As<v8::Function>()->Call(
+            js.v8Context(), value, 0, nullptr));
```

Comment:
The `Symbol.dispose` method is not supposed to return a value, but if is does, I wonder if we should emit a warning here... especially true if user code gets it wrong and has `Symbol.dispose` return a promise.

Reply:
Well, it's actually fine if it returns a promise. The promise will still run.

Note that all JSRPC disposal is async, that is, it always runs after the caller has already moved on, and any exceptions thrown by the disposer do not propagate to the caller. Given that, there's not actually anything inherently wrong with a disposer that performs async tasks.

Reply:
Oh, I know. My concern is largely catching coding bugs.. like exceptions that potentially end up being swallowed and just going nowhere.

```
async [Symbol.dispose]() { throw new Error('boom'); }
```

Certainly not a big deal, just being overly cautious.

Reply:
I believe these exceptions will be reported in the inspector as uncaught exceptions FWIW.